### PR TITLE
Use more ruff rules

### PR DESCRIPTION
**Pull request by ChrisLovering**
This updated our ruff config to select all rules, and disable those I thought weren't useful.

The only breaking changes that I can see will affect our repos is the change to `in_whitelist_check`.

# Draft release notes

## Breaking Changes
All breaking changes are related to making boolean default and boolean-typed positional arguments in function definition keyword args only. The arg names and functions are listed below.
* `sync_app_commands` in `pydis_core.BotBase.load_extensions()`
* `should_raise` in `pydis_core.site_api.APIClient.maybe_raise_for_status()`
* `fail_silently` in `pydis_core.utils.checks.in_whitelist_check()`
* All args other than `pagination_emojis, lines, ctx, embed` in `pydis_core.utils.pagination.LinePaginator.paginate()`

## What's Changed
* Enable more ruff linting rules, by @ChrisLovering in https://github.com/python-discord/bot-core/pull/206
* Document the instance attributes of `pydis_core.BotBase`, by @shtlrs in https://github.com/python-discord/bot-core/pull/204
* Bump ruff from 0.1.15 to 0.2.2, using the new lint config namespace, and linting with the new rules, by @ChrisLovering in https://github.com/python-discord/bot-core/pull/206

**Full Changelog**: https://github.com/python-discord/bot-core/compare/v10.7.0...v11.0.0

### CODE REVIEW

**Review by mbaruh — APPROVED** [review]
The trailing comma rule seems a bit intrusive, and not always relevant (for example after a `**kwargs`), but otherwise this looks good

**Comment by ChrisLovering** [comment]
> The trailing comma rule seems a bit intrusive, and not always relevant (for example after a `**kwargs`), but otherwise this looks good

Yea I can understand that. The linting rule likely makes more sense when used in combination with ruff format / black, as that impacts how they are formatted.

File: tests/pydis_core/test_api.py [hunk 1]
```diff
@@ -17,18 +17,15 @@ def test_response_code_error_default_initialization(self):
         """Test the default initialization of `ResponseCodeError` without `text` or `json`"""
         error = site_api.ResponseCodeError(response=self.error_api_response)
 
-        self.assertIs(error.status, self.error_api_response.status)
-        self.assertEqual(error.response_json, {})
-        self.assertEqual(error.response_text, None)
-        self.assertIs(error.response, self.error_api_response)
+        assert error.status is self.error_api_response.status
```
Comment (shtlrs) [inline tests/pydis_core/test_api.py:20]: Do we want such a change ? the assertion methods offered by unittest do a lot more things than the native one. For instance, comparing lists is easier as it points out the difference between them, etc.

Comment (ChrisLovering) [reply tests/pydis_core/test_api.py:20]: I don't really mind either way, can revert this for now to get it merged

**Pull request merged**